### ISSUE TITLE
Add support for monitoring performance on validation set

### ISSUE THREAD

**Issue by yankang18** [FederatedAI/FATE#302]
It is extremely important for FATE to provide users with functionality on monitoring the performance (on validation set) of the model being trained. For one thing, knowing how the performance is getting progressed while training, users can know their model better. For another, without that mechanism, users might feel frustrated while waiting for the training to end. 

In addition, the early-stopping mechanism should also be provided as it is an important way to regularize the model and avoid overfitting.
